code/game/objects/items/weapons/lighters.dm

```diff
@@ -90,26 +102,38 @@
 		playsound(src, 'sound/items/lighter/plastic_close.ogg', 25, TRUE)
 		next_off_message = world.time + 5 SECONDS
 
-/obj/item/lighter/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
-	if(!isliving(M))
+/obj/item/lighter/attack(mob/living/target, mob/living/user)
+	var/obj/item/clothing/mask/cigarette/cig = locate_cigarette(target)
+	if(istype(cig) || user.a_intent != INTENT_HELP)
```

Comment:
Why not do the normal cigarette_check here?

Reply:
Lighters specifically use a different check so that the entire body can be targeted instead of just the mouth. Since they're _actually_ the intended tools for lighting cigarettes, I thought it'd make sense to make them more convenient to use.